### PR TITLE
Item sub category to revert to "Select an option" when the Item Categ…

### PR DESCRIPTION
**Pull request by Deepak-Bagrawat**
Item sub category to revert to "Select an option" when the Item Category is changed.
https://support.cop.homeoffice.gov.uk/browse/EAB-867

### CODE REVIEW

File: src/Select/Select.jsx [hunk 1]
```diff
@@ -31,6 +31,16 @@ const Select = ({
   const classModifiers = [...toArray(_classModifiers), error ? 'error' : undefined];
   const classes = classBuilder(classBlock, classModifiers, className);
   const [selected, setSelected] = useState(value || defaultValue || '');
+  const [selectAnOption, setSelectAnOption] = useState(options || []);
```
Comment (LewisClark-HO) [inline src/Select/Select.jsx:34]: I think `selectAnOption` isn't a great name for this as it doesn't really describe what the variable is. Maybe something like `cachedOptions` might work better?

File: src/Select/Select.jsx [hunk 2]
```diff
@@ -31,6 +31,16 @@ const Select = ({
...
+
+  useEffect(() => {
+  if(!selectAnOption || selectAnOption?.length !== options?.length ) {
```
Comment (LewisClark-HO) [inline src/Select/Select.jsx:37]: I think `!selectAnOption` here isn't needed as `selectAnOption` should only every be `options` if it's valid, or an empty array (which is a 'truthy' value).

Comment (LewisClark-HO) [inline src/Select/Select.jsx:37]: Also comparing the `length` of the two arrays isn't a great way to do this - what if `options` is a different list but has the same number of entries in it as `selectAnOption`?

**Review by LewisClark-HO — CHANGES_REQUESTED** [review]

File: src/Select/Select.jsx (hunk 2 above)
Comment (Deepak-Bagrawat) [reply src/Select/Select.jsx]: with the length now checking value also.

**Comment by LewisClark-HO** [comment]
Also, some tests are failing after these changes have been made.

File: src/Select/Select.jsx [hunk 3]
```diff
@@ -31,6 +31,22 @@ const Select = ({
   const classModifiers = [...toArray(_classModifiers), error ? 'error' : undefined];
   const classes = classBuilder(classBlock, classModifiers, className);
   const [selected, setSelected] = useState(value || defaultValue || '');
+  const [cachedOptions, setCachedOptions] = useState(useState(options || []));
+
+  useEffect(() => {
+    function checkEqual(cachedOptions, options) {
+      return (
+        cachedOptions?.length === options?.length && 
+        cachedOptions[0]?.id === options[0]?.id &&
```
[the remark this answers is not in the document]
Comment (Deepak-Bagrawat) [reply src/Select/Select.jsx:40]: Yes, from refdata all have id and I checked Terminal, team, sub-location, ports, grade JSON files all of them have id.

[the remark this answers is not in the document]
Comment (Deepak-Bagrawat) [reply src/Select/Select.jsx]: replaced id with value,

**Review by LewisClark-HO — APPROVED** [review]

**Comment by LewisClark-HO** [comment]
LGTM, thanks for making those changes :)

**Pull request merged**